src/main/java/Duke.java

```diff
@@ -1,10 +1,111 @@
+import task.*;
+
+import java.io.*;
+import java.text.ParseException;
+
 public class Duke {
+
+    private static final String FILEPATH = "duke.txt";
+    private Ui ui;
+    private Storage storage;
+    private TaskList taskList;
+
+    /**
+     * Constructor for Duke class.
+     * @param filePath Path for the Duke save file.
+     */
+    public Duke(String filePath) {
+        //storage = new Storage(filePath);
+        taskList = Storage.loadTasks(FILEPATH);
+        ui = new Ui();
+    }
+
+    /**
+     * Method to run the Duke program.
+     */
+    public void run() {
+        Ui.printWelcome();
+        while(true){
+            try {
+                String input = ui.getInput();
+                String[] command = Parser.parseCommand(input);
+
```

Comment:
Perhaps you could try to avoid further deep nesting here? There are already 3 levels of indentation. If you could cut down the levels of indentation further, I believe your code will be much clearner.